README.md

```diff
@@ -1,3 +1,166 @@
+
+Both of these methods use `<Subview>` under the covers, and they require the same props:
+  - `name`: the name of the subview. Allows other subviews to constrain themselves to this subview.
+  - `intrinsicWidth`: the minimum width of the component before application of constraints.
+  - `intrinsicHeight`: the minimum height of the component before application of constraints.
+  - `constraints`: a set of constraints for this subview.
+
+Building constraints uses a fluent DSL in the style of Chai assertions. Some constraint examples:
+
+```es6
+
+// These two constraints center the subview in the superview
+// (the superview being the subview's parent <View> component).
+constrain().subview("demo").centerX.to.equal.superview.centerX
+constrain().subview("demo").centerY.to.equal.superview.centerY
+
+// This prevents this subview from overlapping with
+// another subview named "other".
+constrain().subview("demo").left
+  .to.be.greaterThanOrEqualTo.subview("other").right
+
+// This prevents this subview from overflowing
+// the superview's right edge.
+constrain().subview("demo").right
+  .to.be.lessThanOrEqualTo.superview.right
+```
+
+Here's an example of a custom component using `<Subview>`:
+
+```es6
+class Rectangle extends Component {
+  static propTypes = {
+    layout: PropTypes.shape({
+      top: PropTypes.number,
+      right: PropTypes.number,
+      bottom: PropTypes.number,
+      left: PropTypes.number
+    })
+  };
+
+  render() {
+    return (
+      <div style={{
+        position: "absolute",
+        top: this.props.layout.top || 0,
+        left: this.props.layout.left || 0
+        width: this.props.layout.width || 0,
+        height: this.props.layout.height || 0
+      }}
+      >
+        <p>It's demo time!</p>
+      </div>
+    );
+  }
+}
+
+export default Subview(Rectangle);
+```
+
+Here's how to use `AutoDOM` components:
+
+```es6
+import ConstraintLayout, { View, AutoDOM } from "radium-constraints";
+
+<ConstraintLayout>
+  <View
+    name="tutorial"
+    container="div"
+    width={400}
+    height={500}
+    style={{
+      background: "red"
+    }}
+  >
+    <AutoDOM.div
+      name="tutorialSubview"
+      intrinsicWidth={50}
+      intrinsicHeight={50}
+      constraints={[
+        constrain().subview("tutorialSubview").centerX
+          .to.equal.superview.centerX,
+        constrain().subview("tutorialSubview").centerY
+          .to.equal.superview.centerY
+      ]}
+    >
+      This is a subview
+    </AutoDOM.div>
+  </View>
+</ConstraintLayout>
+```
+
+When using `AutoSVG` components, make sure to pass "g" instead of "div" to the `<View>`'s `container` prop.
+
+## Demo
+There are more complex examples on the demo page. Check out the code in [app.jsx](https://github.com/FormidableLabs/radium-constraints/blob/master/demo/app.jsx).
+
+### Installation
+- Install builder: `npm install -g builder`
```

Comment:
Please, please don't recommend this. Also, as a guiding principle, (almost) never do global installs.

We have a standard blurb at the top of this document that you can copy and paste in: https://github.com/FormidableLabs/builder-react-component/blob/master/DEVELOPMENT.md